src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs

```diff
@@ -50,10 +50,12 @@ public Task<bool> TryNavigateToItemAsync(Document document, NavigationBarItem it
         protected async Task NavigateToPositionAsync(Workspace workspace, DocumentId documentId, int position, int virtualSpace, CancellationToken cancellationToken)
         {
             var navigationService = workspace.Services.GetRequiredService<IDocumentNavigationService>();
-            if (await navigationService.CanNavigateToPositionAsync(workspace, documentId, position, virtualSpace, cancellationToken).ConfigureAwait(false))
+            var location = await navigationService.GetLocationForPositionAsync(
+                workspace, documentId, position, virtualSpace, NavigationOptions.Default, cancellationToken).ConfigureAwait(false);
```

Comment:
This randomly jumped out to me now: should the NavigationOptions be passed to NavigateTo for the location rather than consuming it here? It seems odd that we're passing it only to more or less curry it to that call later. I recognize the shape is partly because of how the split was implemented, but something to think about.

This can be left for a future PR since I know you have plenty on the way.

Reply:
yes.  that seems more sensible. t hanks!

Reply:
i've made this change. a nd it cleans things up.  awesome!